### ISSUE TITLE
Relationships are not mappend if defined in abstract class constructor.

### ISSUE THREAD

**Issue by meistermeier** [spring-projects/spring-data-neo4j#2452]
This is related to the already solved #2418

A relationship in will not get mapped if it is defined in an abstract class and queried through a repository defined for the abstract class.

Example:
```
abstract class AbstractClass {
@Relationship
public Something something;
}
```
and
```
class ConcreteClass extends AbstractClass {
}
```
with a repository
```
interface AbstractClassRepository extends Neo4jRepository<AbstractClass,...>
```

Will create a query containing `AbstractClass_SOMETHING_something` for the relationship.
On the mapping side, the code already figured out that the class to create an instance from is `ConcreteClass` and it will look for a `ConcreteClass_SOMETHING_something` defined property containing the relationship results.

In conclusion the mapping part should always check if there might be an entry for the more abstract type(s) above in the result list.

**Comment by meistermeier** [comment]
Additional information: This bit is only missing when using relationships in the constructor. #2418 already solved the property (Setter/Wither) problem.
